### DISCUSSION TITLE
Concurrent issue with `helm dependency build` in argocd-repo-server

### DISCUSSION THREAD

**Discussion by arielly-parussulo** [argoproj/argo-cd#12877] [Q&A]
### The issue
I have a question about an issue that I found in the **argocd-repo-server** logs for my company's applications:

```
helm dependency build` failed exit status 1: Error: could not download https://my-repo-url/chart-version.tgz: no cached repo found. (try 'helm repo update'): error loading /helm-working-dir/repository/https:--***-http-index.yaml: empty index.yaml file" grpc.code=Unknown grpc.method=GenerateManifest grpc.service=repository.RepoServerService grpc.start_time="2023-03-15T15:42:36Z" grpc.time_ms=4666.771 span.kind=server system=grpc
```

By checking issues in the [helm repository](https://github.com/helm/helm/issues/10735) I found out that this issues happens because repo-server executes processes in parallel to generate the helm manifests and by running the command `helm dependency build`, it tries to recreate the index.yaml cached file of this Helm repository at the same time of another process. 

### Possible solution
The helm repository thread mentions the parameter `--repository-cache` to use different local indexes to fix this issue. I didn't like the idea because we have lots of applications and this command would probably use a lot of disk space as it would replicate the cache files for each application. Also I didn't find a way of passing this parameter for Helm in ArgoCD.
My question is if anybody already faced this issue and if it was found a way of solving this. 

#### ArgoCD version
We are using argocd **2.4.11**.

**Comment by crenshaw-dev** [comment]
I've created an issue to track this. Feels like we probably need a locking mechanism on the Helm side. https://github.com/argoproj/argo-cd/issues/12902
